app/models/measurement_details.rb

```diff
@@ -21,12 +21,17 @@ def initialize(attributes = {})
   def load
     validate!
 
-    load_total_from_gr
-    load_local_from_gr
-    load_user_from_gr
-    load_sub_mins_from_gr
-    load_team_from_gr
-    load_split_measurements
+    # make sure instance vars are set
+    @current_power = Power.current
+    gr_singleton
+    ministry
+
+    threads = [:load_local_from_gr, :load_total_from_gr, :load_user_from_gr, :load_sub_mins_from_gr,
+               :load_team_from_gr, :load_split_measurements]
+    threads = threads.map do |method|
+      Thread.new { send(method) }
+    end
+    threads.each { |t| t.join(1) }
```

Comment:
I actually don't think I know enough to make a good suggestion. My guess is that both the threads getting hung and the requests taking longer than 6 seconds would be rare events. Actually the threads would continue to run in the background after the timeout eating resources if a bunch got hung.

So I think my cautious suggestion based on what I do know is to either bump the timeout up a bit and add a comment or just remove the timeout and if the requests hang in practice to revisit this.